browser/src/control/Control.ContextToolbar.ts

```diff
@@ -119,6 +119,11 @@ class ContextToolbar extends JSDialogComponent {
 			this.container.style.left = pos.x + 'px';
 			this.container.style.top = pos.y + 'px';
 			window.L.DomUtil.removeClass(this.container, 'hidden');
+			if (this.container.getBoundingClientRect().right > window.innerWidth) {
```

Comment:
We assign here `this.container.style.left`, then use getBoundingClientRect() which requires to relayout and calculate, then in case of need we again setup new `this.container.style.left` which will require another layouting.

Can we squash it?
- calculate first, set once ?

Also we could call `getCanvasBoundingClientRect` above only once and store in variable.

Reply:
could you check the perf profile? if it doesn't trigger too long layouting ?
best to compare both

Reply:
I might be wrong, just assumed that from code